safe-transaction-client/src/lib.rs

```diff
@@ -89,11 +100,42 @@ impl Client<'_> {
         }
     }
 
-    pub fn with_safe(&self, safe_address: Address) -> Safe {
-        Safe {
+    pub fn get_safe(&self, safe_address: Address) -> Result<Safe, Error> {
+        #[derive(serde::Deserialize)]
+        struct SafeResponse {
+            nonce: u64,
+            threshold: u64,
+            owners: Vec<String>,
+        }
+
+        let SafeResponse {
+            nonce,
+            threshold,
+            owners,
+        } = self
+            .agent
+            .get(&format!(
+                "{}/v1/safes/{}/",
+                self.transactions_api,
+                to_checksum(&safe_address, None),
+            ))
+            .call()?
+            .into_json()?;
+
+        let owners = owners
+            .iter()
+            .map(|o| Address::from_str(o))
```

Comment:
You may use `.map(Address::from_str)`.

Reply:
Actually this is a case where you can't for some annoying reason:

```rust
error[E0631]: type mismatch in function arguments
   --> safe-transaction-client/src/lib.rs:115:18
    |
115 |             .map(Address::from_str)
    |                  ^^^^^^^^^^^^^^^^^
    |                  |
    |                  expected signature of `fn(&std::string::String) -> _`
    |                  found signature of `for<'r> fn(&'r str) -> _`
```